src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java

```diff
@@ -0,0 +1,162 @@
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a domain, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutProject() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, without providing projectId
+    @Test
+    public void testGetCloudUserUnscoped() throws FogbowException {
+        // setup
+        OpenStackV3User fakeToken = new OpenStackV3User(ANY_VALUE, ANY_VALUE, ANY_VALUE, null);
+        mockServices(fakeToken);
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+
+        // verify
+        Mockito.verify(this.plugin).getUnscopedCloudUserFromJson(Mockito.any());
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, providing projectId
+    @Test
+    public void testGetCloudUserScoped() throws FogbowException {
+        // setup
+        OpenStackV3User fakeToken = new OpenStackV3User(ANY_VALUE, ANY_VALUE, ANY_VALUE, ANY_VALUE);
+        mockServices(fakeToken);
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+
+        // verify
+        Mockito.verify(this.plugin).getCloudUserFromJson(Mockito.any());
+    }
+
+    private void mockServices(OpenStackV3User openStackV3User) throws FogbowException {
+        HttpResponse response = new HttpResponse(ANY_VALUE, HttpStatus.OK.value(), new HashMap<>());
+        PowerMockito.mockStatic(HttpRequestClient.class);
+        BDDMockito.when(HttpRequestClient.doGenericRequest(Mockito.anyObject(), Mockito.anyString(), Mockito.anyObject(), Mockito.anyObject()))
+                .thenReturn(response);
+
+        Mockito.doReturn(openStackV3User).when(this.plugin)
+                .getCloudUserFromJson(response);
+
+        Mockito.doReturn(openStackV3User).when(this.plugin)
+                .getUnscopedCloudUserFromJson(response);
+    }
+
+    private Map<String, String> createCredentials(String username, String password, String domain) {
+        Map<String, String> credentials = new HashMap<String, String>();
+
+        credentials.put(OpenStackConstants.Identity.USER_NAME_KEY_JSON, username);
+        credentials.put(OpenStackConstants.Identity.PASSWORD_KEY_JSON, password);
+        credentials.put(OpenStackConstants.Identity.DOMAIN_KEY_JSON, domain);
+
+        return credentials;
+    }
+
```

Comment:
is it possible to use this 4 parameters method and remove the above. In this case, pass null objects when testing the cases of missing parameters

Reply:
Done.